### PR TITLE
Support resultSet direct push for SparkSqlExecutor

### PR DESCRIPTION
**Pull request by ycycse**
Using direct push of result sets, data does not need to be cached on disk and can be calculated in real-time and returned directly from memory. This PR supports direct pushing of resultSets under SparkSqlExecutor.

### CODE REVIEW

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/utils/ArrowUtils.scala [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          vector
+        case BooleanType =>
+          val vector = new BitVector(field.name, allocator)
+          vector.allocateNew(df.count().toInt)
+          vector
+        case StringType =>
+          val vector = new VarCharVector(field.name, allocator)
+          vector.allocateNew(df.count().toInt)
+          vector
+        case _ => throw new IllegalArgumentException(s"Unsupported data type: ${field.dataType}")
```
[the remark this answers is not in the document]
Comment (ycycse) [reply linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/utils/ArrowUtils.scala:82]: You're right, I will improve this part.

File: linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkSqlExecutor.scala [hunk 2]
```diff
@@ -47,6 +44,35 @@ class SparkSqlExecutor(sparkEngineSession: SparkEngineSession, id: Long)
...
+    val dataFrameResponse = DirectPushCache.fetchResultSetOfDataFrame(taskId, fetchSize)
+    if (dataFrameResponse.dataFrame != null) {
+      if (!dataFrameResponse.hasMoreData) {
+        succeedTasks.increase()
+      }
+      FetchResultResponse(
+        hasMoreData = dataFrameResponse.hasMoreData,
+        ArrowUtils.toArrow(dataFrameResponse.dataFrame)
+      )
+    }
```
Comment (peacewong) [inline linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkSqlExecutor.scala:72]: Else is missing here

Comment (ycycse) [reply linkis-engineconn-plugins/spark/src/main/scala/org/apache/linkis/engineplugin/spark/executor/SparkSqlExecutor.scala]: done. After removing the  job-way to fetch data. This method isn't needed anymore.

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**